From 0eda9ca02d5ca7293b224191c785795e6c7e565d Mon Sep 17 00:00:00 2001
From: mytja <mitja@severkar.eu>
Date: Thu, 26 Dec 2024 13:53:22 +0100
Subject: [PATCH 5/5] fix: allow accessing sysfs by SystemUI & shell

This allows NetworkTraffic class in SystemUI to obtain kernel
network traffic statistics from /sys/class/net/*/statistics/{tx/rx}_bytes
even when SELinux is enforcing.

"search" permission is apparently required on 4.14 kernels

Change-Id: I94e88e2496b841530362275861f4ef6b4e5644c3
---
 sepolicy/sysfiles.te | 4 ++++
 1 file changed, 4 insertions(+)
 create mode 100644 sepolicy/sysfiles.te

diff --git a/sepolicy/sysfiles.te b/sepolicy/sysfiles.te
new file mode 100644
index 0000000..5ac28ad
--- /dev/null
+++ b/sepolicy/sysfiles.te
@@ -0,0 +1,4 @@
+allow shell sysfs:file { read open };
+allow platform_app sysfs:file { read open getattr };
+allow platform_app sysfs_net:dir { read open getattr search };
+allow platform_app sysfs:dir { read open getattr search };
-- 
2.34.1

